pkg/goDB/info/hostid_linux.go

```diff
@@ -0,0 +1,25 @@
+//go:build linux
+// +build linux
+
+package info
+
+import (
+	"io"
+	"os"
+)
+
+const machineIDFile = "/etc/machine-id"
```

Comment:
@fako1024 : need help here. Or advise. Or just a comment. I know that this file exists on _some_ distributions. Are you aware of other places on the system that provides equivalent information (in other distributions)?

And in any case: what's your take on this?

Reply:
IMHO it's a bit dangerous. :-P Aside from the fact that it seems bazooka-style to required OS / Arch specific implementations to create a random host id I'm also afraid `/etc/machine-id` won't even cut it across all Linux systems (seems like some Linux derivatives use `/var/lib/dbus/machine-id` (see e.g. this package, which hasn't received any updates in the last years, but still indicates it's not as trivial: https://github.com/denisbrodbeck/machineid).
How about:

- Just using the FQDN of the machine (which should be universally doable)
- If it should be anonymous, use a secure hash on top
- If you're concerned about the FQDN not always being _really_ unique, how about simply generating a unique identifier as part of the goDB (e.g. upon first write to a fresh DB, or upon first distributed query via the API, or ...)

Reply:
Mmmh. Okay. How about we use https://github.com/denisbrodbeck/machineid then? It seems good enough for what we aim to achieve.

Anonymity IMHO is less of a concern that uniqueness. The reason why the unique ID is important, is when service discovery comes into play, e.g. `goProbe` instances registering themselves (on behalf of the host they run on) as reachable API endpoints for global queries.

Not that this system is in place already, but it would make sense to have a unique identifier for the hosts then.

Reply:
TBH I'm not too keen on using a package that has last seen love >5yrs ago (it doesn't even have a `go.mod` yet). What about the last suggestion of simply creating your own identifier and persist it in the DB. Seems to me like:

- That's independent of OS / architecture
- Is intrinsic and has no external dependencies
- Is cheap (take your pick when to do it, but it's one file)

On second thought I still don't fully see why you can't use the FQDN. If it's about host discovery, why not just use, well, the hostname for that? :thinking:

Reply:
Ok with you if we keep current way and tackle this in separate issue? Needs a more involved discussion.